pkg/apis/clusteroperator/v1alpha1/types.go

```diff
@@ -741,3 +742,58 @@ type CombinedCluster struct {
 	ClusterStatus         *capiv1.ClusterStatus
 	ClusterProviderStatus *ClusterProviderStatus
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DNSZone represents a Domain Name Service Zone to manage
+type DNSZone struct {
+	// +optional
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
+	Spec DNSZoneSpec `json:"spec,omitempty"`
+	// +optional
+	Status DNSZoneStatus `json:"status,omitempty"`
+}
+
+// DNSZoneSpec is a specification of a Domain Name Service Zone. The specification will
+// be specific to each cloud provider.
+type DNSZoneSpec struct {
+	// Zone is the DNS zone to host
+	DNSZone string `json:"dnszone"`
```

Comment:
That is very funny, I hadn't thought of the inception there.

It's a string, so the code isn't in an object reference cycle or anything, but conceptually I can see how this would be confusing.

This field needs to exist here as it's the actual "something.example.com" zone that needs to be created. Some places have called it "Name" but I think that's a little confusing since the object metadata also has "Name".

I could rename it to just "zone". Would that suffice do you think?

Reply:
Hah, my mistake I totally misread the type there. DNSZone or Zone or Name or ZoneName, all ok with me.